torch/export/_trace.py

```diff
@@ -2027,6 +2035,43 @@ def _produce_guards_callback(gm):
     )
 
 
+def emit_bogus_const_warning(constants, gs, gm):
+    bogus_constants: set[str] = set()
+    for const, val in constants.items():
+        if isinstance(
+            val, torch._subclasses.fake_tensor.FakeTensor
+        ) and _is_bogus_const_name(const):
+            bogus_constants.add(const)
+
+    if len(bogus_constants) == 0:
+        return
+
+    bogus_constant_names: set[str] = set()
+    for inp in gs.input_specs:
+        if inp.kind == InputKind.CONSTANT_TENSOR and inp.target in bogus_constants:
+            bogus_constant_names.add(inp.arg.name)
+
+    placeholders = {
+        node.name: node for node in gm.graph.nodes if node.op == "placeholder"
+    }
+    for name in bogus_constant_names:
+        placeholder_node = placeholders[name]
+        dependencies: list[str] = []
+        for user in placeholder_node.users:
+            if user.meta.get("stack_trace", None) is not None:
+                dependencies.append(user.meta["stack_trace"])
+        if len(dependencies) > 0:
+            warnings.warn(
```

Comment:
You mentioned in the summary that some internal users complained, but isn't this a pretty bad situation to run into?

Can you think of a case where tracing was done with fake tensors that could potentially cause this situation to be hit?

Reply:
Yeah in their case, constants didn't show up in the graph so it was kinda okay. We still need to get repro from them and fix it properly but they were pretty persistent about it being not fatal error for them.

Reply:
ok after thinking about it bit more, i think this situation is almost always gonna be pretty bad. We should probably just error and disable the error on their side. I feel it would be good to let ppl complain to us when they hit this error, so that we actually find corner cases in our tracing system.